src/middlewared/middlewared/api/base/types/user.py

```diff
@@ -0,0 +1,25 @@
+import string
+
+from pydantic.functional_validators import AfterValidator
+from typing_extensions import Annotated
+
+__all__ = ["LocalUsername"]
+
+
+def validate_local_username(val):
```

Comment:
I'm a bit torn on how this is structured. This function is in a file called `api/base/types/user.py`. That's not very intuitive to me. I wonder if we need to have a directory structure where we put functions in their own files and keep the "types" directory only filled with files that have type definitions/annotations?

Reply:
Is it important where this function is? It's not meant to be used publicly (`__all__` only contains `LocalUsername`)